src/node_http2_state.h

```diff
@@ -135,7 +139,8 @@ class Http2State : public BaseObject {
     double stream_stats_buffer[IDX_STREAM_STATS_COUNT];
     double session_stats_buffer[IDX_SESSION_STATS_COUNT];
     uint32_t options_buffer[IDX_OPTIONS_FLAGS + 1];
-    uint32_t settings_buffer[IDX_SETTINGS_COUNT + 1];
+    uint32_t settings_buffer[IDX_SETTINGS_COUNT + 1 + 1 +
+                             2 * MAX_ADDITIONAL_SETTINGS];
```

Comment:
nit: a comment about what the additional ` + 1 + 2 ...` is coming from would be helpful.

Reply:
Would
```
    // first + 1: number of actual nghttp2 supported settings
    // second + 1: number of additional settings not suppoted by nghttp2
    // 2 * MAX_ADDITIONAL_SETTINGS: settings id and value for each
    // additional setting
```
be ok?

Reply:
perfect!